Task 8

dkozhemyako opened the pull request.

dkozhemyako committed:
Merge pull request #5 from dkozhemyako/task-7

task 7 -> task 8

dkozhemyako committed:
task 8 in progress

dkozhemyako committed:
task 8

v1.0

alexpalyan commented on add.php:
```diff
@@ -0,0 +1,74 @@
+  
+    if ($taskAdd === true)
+      {
+        header("Location: http://localhost/hillel-homework/index.php");
```
домен может быть абсолютно разным и приложение должно уметь работать ) Лучше редирект делать на сам файл сценария index.php

еще важный момент - базовый путь тоже может быть разным... тут либо нужно настроить чтобы приложение работало в корне по адресу http://localhost/index.php либо вынести это в конфигурацию...

alexpalyan commented on add.php:
```diff
@@ -0,0 +1,74 @@
+if (isset($_POST['btn_task_add']))
+{
+  $args = array(
+    'inputName'   => FILTER_SANITIZE_STRING,
```
Используйте валидационные фильтры или фильтр по умолчанию, который не будет обрабатывать строку. еще желательно прописать критерии - допустим, максимальная длина строки, которая может быть записана в базу )

alexpalyan commented on add.php:
```diff
@@ -0,0 +1,74 @@
+
+  if (isDateValid($myinputs['inputDate']) === false || differenceDateH($myinputs['inputDate'], '')['calc'] == 0)
+  {
+    $input_errors += ['inputDate' => 'Вкажіть коректну дату'];
```
установку значений по ключам элементов в массив лучше делать через
$input_errors['inputDate'] = 'Вкажіть коректну дату';

alexpalyan commented on add.php:
```diff
@@ -0,0 +1,74 @@
+
+  if (checkProjects($myinputs['selectProject'], $projects) === false)
+  {
+    $input_errors += ['selectProject' => 'Вкажіть існуючий проект'];
```
установку значений по ключам элементов в массив лучше делать через
$input_errors['selectProject'] = 'Вкажіть існуючий проект';

alexpalyan commented on add.php:
```diff
@@ -0,0 +1,74 @@
+
+  if ($myinputs['inputName'] === '')
+  {
+    $input_errors += ['inputName' => 'Вкажіть назву задачі'];
```
установку значений по ключам элементов в массив лучше делать через
$input_errors['inputName'] = 'Вкажіть назву задачі';

alexpalyan commented on templates/task-add.php:
```diff
@@ -111,9 +111,9 @@ <h1>Додати проект</h1>
         </div>
       </section>
-
+      
```
Шаблон должен содержать только секцию контента )

alexpalyan commented on templates/card.php:
```diff
@@ -2,9 +2,10 @@
                   <h5 class="card-title"><?=htmlentities($task['name'])?></h5>
                   <div class="card-tools">
-                    <a href="#" class="btn btn-tool btn-link">#3</a>
+                    <a href="<?=$task['data_set']?>" class="btn btn-tool btn-link">file</a>
```
Это ссылка не для файла ) она - номер задачи ) Но пусть будет )

alexpalyan commented on templates/card.php:
```diff
@@ -2,9 +2,10 @@
                     <a href="#" class="btn btn-tool">
-                      <i class="fas fa-pen"></i>
+                      <i class="fas fa-pen">
+                      </i>
```
Тут важно чтобы небыло переноса ) иначе может неверно отображаться )

dkozhemyako committed:
task-8 v1.1

dkozhemyako committed:
task-8 update

dkozhemyako committed:
task-8

cs
php 8.1 errors

dkozhemyako committed:
task-8

add change

dkozhemyako committed:
task 8

del '' in textarea

dkozhemyako committed:
task-8

 - date is not a required field
- strlen filds
- textarea is invalid
- data_set if empty

dkozhemyako committed:
Merge branch 'main' into task-8
